internal/diagnosticwriter/diagnosticwriter.go

```diff
@@ -370,15 +455,15 @@ func prettyPathForFileError(file *ast.SourceFile, fileErrors []*ast.Diagnostic,
 	)
 }
 
-func WriteFormatDiagnostics(output io.Writer, diagnostics []*ast.Diagnostic, formatOpts *FormattingOptions) {
+func WriteFormatDiagnostics(output io.Writer, diagnostics []Diagnostic, formatOpts *FormattingOptions) {
 	for _, diagnostic := range diagnostics {
 		WriteFormatDiagnostic(output, diagnostic, formatOpts)
 	}
 }
 
-func WriteFormatDiagnostic(output io.Writer, diagnostic *ast.Diagnostic, formatOpts *FormattingOptions) {
+func WriteFormatDiagnostic(output io.Writer, diagnostic Diagnostic, formatOpts *FormattingOptions) {
 	if diagnostic.File() != nil {
-		line, character := scanner.GetECMALineAndCharacterOfPosition(diagnostic.File(), diagnostic.Loc().Pos())
+		line, character := scanner.GetECMALineAndCharacterOfPosition(diagnostic.File(), diagnostic.Pos())
```

Comment:
We haven't really vetted these calls yet, but I am worried that this is going to be wrong given this func accepts AST diags (UTF-8 offsets), LSP diags (UTF-16 by default?), and then `GetECMALineAndCharacterOfPosition` returns a rune count (UTF-32???) up to `Pos` for some reason.

We can probably kick the can down the road if it's not _crashing_ yet, but, yeah